src/main/java/org/reactome/release/downloaddirectory/GenerateGOAnnotationFile/CellularComponentAnnotationBuilder.java

```diff
@@ -2,88 +2,180 @@
+    public static List<String> processCellularComponents(GKInstance reactionlikeEvent) throws Exception {
         List<String> goaLines = new ArrayList<>();
         // First retrieve proteins, then build GO annotation
-        for (GKInstance proteinInst : GOAGeneratorUtilities.retrieveProteins(reactionInst)) {
-            GKInstance referenceEntityInst = (GKInstance) proteinInst.getAttributeValue(ReactomeJavaConstants.referenceEntity);
-            GKInstance speciesInst = (GKInstance) proteinInst.getAttributeValue(ReactomeJavaConstants.species);
+        for (GKInstance protein : retrieveProteins(reactionlikeEvent)) {
             // Check if the protein has any disqualifying attributes.
-            boolean validProtein = GOAGeneratorUtilities.isValidProtein(referenceEntityInst, speciesInst);
-            if (validProtein) {
-                String taxonIdentifier = ((GKInstance) speciesInst.getAttributeValue(ReactomeJavaConstants.crossReference)).getAttributeValue(ReactomeJavaConstants.identifier).toString();
-                // For CC annotations, alternateGOCompartments are disqualifying.
-                if (!GOAGeneratorUtilities.isExcludedMicrobialSpecies(taxonIdentifier)) {
-                    if (!speciesWithAlternateGOCompartment.contains(taxonIdentifier)) {
-                        // For CC, we are looking at ALL proteins in a ReactionlikeEvent
-                        goaLines.add(generateGOCellularCompartmentLine(proteinInst, reactionInst, taxonIdentifier));
-                    } else {
-                        logger.info("Species has an alternative GO compartment, skipping GO annotation");
-                    }
-                } else {
-                    logger.info("Protein is from an excluded microbial species, skipping GO annotation");
+            String issueDisqualifyingProtein = getAnyIssueForCellularComponentDisqualification(protein);
+            if (issueDisqualifyingProtein.isEmpty()) {
+                String goaLine = generateGOCellularCompartmentLine(protein, reactionlikeEvent);
+                if (!goaLine.isEmpty()) {
+                    goaLines.add(goaLine);
                 }
             } else {
-                logger.info("Invalid protein, skipping GO annotation");
+                logger.warn(issueDisqualifyingProtein);
             }
         }
         return goaLines;
     }
 
     /**
-     * Proteins reaching this part of the script are deemed valid for GO annotation. Retrieves the Cellular Compartment accession associated with this instance and then calls the GOA line generator.
-     * @param proteinInst -- GKInstance, Individual protein instance from the retrieved proteins.
-     * @param reactionInst -- GKInstance, Parent reaction that the protein is associated with.
-     * @param taxonIdentifier -- String, CrossReference ID of protein's species.
+     * Performs an AttributeQueryRequest on the incoming reaction instance. This will retrieve all protein's
```

Comment:
Spelling: "protein's" should be "proteins"

Reply:
Changed!